install/update_94_95.php

```diff
@@ -927,6 +927,50 @@ function update94to95() {
    }
    /** /Fix mixed classes case in DB */
 
+   /** A doc_item to rule them all! */
+   $itemtypes = [
+      'ITILFollowup' => 'content',
+      'ITILSOlution' => 'content',
+      'Reminder'     => 'text',
+      'KnowbaseItem' => 'answer'
+   ];
+   foreach (['Change', 'Problem', 'Ticket'] as $itiltype) {
+      $itemtypes[$itiltype] = 'content';
+      $itemtypes[$itiltype . 'Task'] = 'content';
+   }
+
+   $docs_input =[];
+   foreach ($itemtypes as $itemtype => $field) {
+      // Check ticket and child items (followups, tasks, solutions) contents
+      $regexPattern = 'document\\\.send\\\.php\\\?docid=[0-9]+';
+
+      $result = $DB->request([
+         'SELECT' => ['id', $field],
+         'FROM'   => $itemtype::getTable(),
+         'WHERE'  => [
+            $field => ['REGEXP', $regexPattern]
+         ]
+      ]);
+
+      while ($data = $result->next()) {
+         //TODO create Document_Item entry, with a timeline_position set to Document_Item::NO_TL
+         preg_match('/document\\.send\\.php\\?docid=([0-9]+)/', $data[$field], $matches);
+         $docs_input[] = [
+            'documents_id' => $matches[1],
+            'itemtype'     => $itemtype,
```

Comment:
For ITILFollowup and ITILSolution, I think `'itemtype'` should be the itemtype of the parent ITIL item.

Reply:
Dunno... ping @orthagh

Reply:
I do not agree.
Doing this, this would be impossible to link image from its original real place (say a Fup); while it is possible to link it currently to the parent itemtype throw the Fup.

Reply:
This is not consistent with what will be done when a document will be added from the timeline form (see https://github.com/glpi-project/glpi/pull/6598/files#diff-7f569484c496f62a6ccf54647fed2b1bR5070).

Another problem if we do not use the parent ITIL item is that `ITILFollowup` and `ITILSolution` are not defined in `$CFG_GLPI['document_types']`.

Consequences of this are that `Document::getItemtypesThatCanHave()` will not list them and `Document::canApplyOn()` will return false for them:
 - it may have an impact on transfer process,
 - cleaning process in `CommonDBTM::cleanRelationTable()` will not be able to clean `document_items` table properly.

If we declare them in `$CFG_GLPI['document_types']`, I do not see any other consequence than having ability to manually link an existing document to a followup or a solution.

Reply:
OK, I've changed in commondbtm to have the same behavior on both places; and I've added possible missing types in CFG_GLPI